src/gsi/read_satwnd.f90

```diff
@@ -413,221 +350,37 @@ subroutine read_satwnd(nread,ndata,nodata,infile,obstype,lunout,gstime,twind,sis
         endif
             
         call ufbint(lunin,hdrdat,13,1,iret,hdrtr_v1) 
+        if(hdrdat(9)>=10.d10) call ufbint(lunin,hdrdat(9),1,1,iret,'CMCM')
           ! SWQM doesn't exist for GOES-R/new BUFR/ hence hdrdat(13)=MISSING.
           ! qm=2, instead of using hdrdat(13)(2015-07-16, Genkova)
 
-        iobsub=0
-        itype=-1
-        iobsub=int(hdrdat(1))
+        isaid=nint(hdrdat(1))
         ihdr9=nint(hdrdat(9))
+        if(ihdr9<1.or.ihdr9>7) cycle loop_report 
 
-        if(istype == 1) then
-           if( hdrdat(1) <r80 .and. hdrdat(1) >= r50) then   !the range of EUMETSAT satellite IDS
-              if(ihdr9 == 1)  then                           ! IR winds
-                 itype=253
-              else if(ihdr9 == 2) then                       ! visible winds
-                 itype=243
-              else if(ihdr9 == 3) then                       ! WV cloud top
-                 itype=254
-              else if(ihdr9 >= 4) then                       ! WV deep layer, monitored
-                 itype=254
-              endif
-           endif
+        itype   = sattab(ysub,isaid,ihdr9,1)
+        istype  = sattab(ysub,isaid,ihdr9,2)
 
-        else if(istype == 2) then               ! read new EUM BURF
-           if( hdrdat(1) <r80 .and. hdrdat(1) >= r50) then   !the range of EUMETSAT satellite IDS
-              if(ihdr9 == 1)  then                           ! IR winds
-                 itype=253
-              else if(ihdr9 == 2) then                       ! visible winds
-                 itype=243
-              else if(ihdr9 == 3) then                       ! WV cloud top
-                 itype=254
-              else if(ihdr9 >= 4) then                       ! WV deep layer, monitored 
-                 itype=254
-              endif
-           endif
+        if (istype == -1) cycle loop_report ! unassigned itypes
+        if (itype  == -1) write(6,*) 'type mismatch', itype,istype
+        if (itype  == -1) cycle loop_report ! unassigned itypes    `
```

Comment:
Can we combine lines 365 and 366 into a single if block
```
if (itype  == -1) then
   write(6,*) 'type mismatch', itype,istype
   cycle loop_report ! unassigned itypes
endif
```

We should remove the extra white space including the ` character in the comment on line 366.  Instead of 
```
cycle loop_report ! unassigned itypes    `
```
we should have 
```
cycle loop_report ! unassigned itypes 
```

Reply:
Done.